test/testutil.h

```diff
@@ -141,10 +143,94 @@ __owur int run_tests(const char *test_prog_name);
  */
 
 /*
- * Returns 1 if |s1| and |s2| are both NULL or equal.
- * Otherwise, returns 0 and pretty-prints diagnostics using |desc|.
+ * A common routine to output test failure messages.  Generally this should not
+ * be called directly, rather it should be called by the following functions.
+ *
+ * |desc| is a printf formatted description with arguments |args|, |type| is
+ * the data type that was tested, |fmt| is a system provided printf format with
+ * arguments that spells out the failure details.
+ */
+#if defined(__GNUC__)
```

Comment:
I'm assuming that the output of fail_message is in a dreadful format and will be changed.  I'll add better descriptions as to what its arguments are and do.